renderer/canvas.py

```diff
@@ -271,18 +291,23 @@ def _draw_food(self, surface, obs):
         while len(stack_list) > 0:
             i = 0
             while i < len(stack_list):
-                food_above, food_below = stack_list[i]
-                if food_below in stack_number:
+                item_above, item_below = stack_list[i]
+                if item_below in stack_number:
                     stack_list.remove(stack_list[i])
-                    stack_number[food_above] = stack_number[food_below] + 1
+                    stack_number[item_above] = stack_number[item_below] + 1
                     # Get location of station
                     for literal in obs:
-                        if literal.predicate == "at" and literal.variables[0].name == food_below:
+                        if literal.predicate == "at" and literal.variables[0].name == item_below:
                             station_pos = self._get_station_position(literal.variables[1].name)
                             break
-                    cheese_offset = -0.05 if "cheese" in food_above or "onion" in food_above else 0
-                    station_pos[1] -= self.STATION_FOOD_OFFSET + 0.1 * (stack_number[food_above] - 1) + cheese_offset
-                    self._draw_food_image(surface, food_above, obs, station_pos * self.pix_square_size)
+                    stack_offset = 0
+                    for item in item_above:
```

Comment:
This doesn't make sense; `item_above` is a string so you are iterating through the characters (see line 297). Let's say you fix that and you just have the try-except, try excepts are ugly so let's try to avoid using it. Instead, attempt to do
```
stack_offset = self.config["items"]["entities"][item_above]["constants"].get("STACK_OFFSET", 0)
```
to safely attempt to retrieve the stack offset constant if it exists or set it to 0 if not. This is much more readable than an unnecessary try-except.